blockprod/src/detail/mod.rs

```diff
@@ -452,8 +464,13 @@ impl BlockProduction {
                     .await?;
             }
 
-            let (consensus_data, block_reward, current_tip_index, finalize_block_data) =
-                self.pull_consensus_data(input_data.clone(), self.time_getter.clone()).await?;
+            let (
+                consensus_data,
+                block_reward,
+                current_tip_index,
+                median_time_past,
```

Comment:
Can we call this `curr_best_median_time_past`?

We can add documentation to this stating that "this is a conservative estimate of the median time point, at which locked transactions can be unlocked and other related calculations can be done.

Reply:
> Can we call this curr_best_median_time_past?
We can add documentation to this stating that "this is a conservative estimate of the median time point, at which locked transactions can be unlocked and other related calculations can be done.

I've named it `current_tip_median_time_past`. The comment is also formulated a bit differently, let me know if you have any objections.